### PR TITLE
Grid removal

### LINKED ISSUE #105: Remove grid
It has been decided that the UI should not include a grid, as this does not provide any meaningful value to the user. Instead, a partly opaque background should be used for the canvas

### PR DESCRIPTION
**Pull request by Nielswps**
This removes the Grid from the editor, lowering the amount of computations needed as well as the snapping of elements

Closes #105

### CODE REVIEW

File: src/main/java/ecdar/abstractions/ComponentInstance.java [hunk 1]
```diff
@@ -11,8 +9,8 @@
  * Instance of a component.
  */
 public class ComponentInstance implements SystemElement {
-    public final static int WIDTH = Grid.GRID_SIZE * 22;
-    public final static int HEIGHT = Grid.GRID_SIZE * 12;
+    public final static int WIDTH = 220;
+    public final static int HEIGHT = 120;
```
Comment (seblund) [inline src/main/java/ecdar/abstractions/ComponentInstance.java:13]: Consider making this a function of the canvas padding:
```suggestion
    public final static int WIDTH = Ecdar.CANVAS_PADDING * 22;
    public final static int HEIGHT = Ecdar.CANVAS_PADDING * 12;
```

File: src/main/java/ecdar/abstractions/SystemRoot.java [hunk 2]
```diff
@@ -33,12 +33,12 @@ public DoubleProperty getXProperty() {
...
     @Override
     public ObservableValue<? extends Number> getEdgeX() {
-        return getXProperty().add(3 * Grid.GRID_SIZE);
+        return getXProperty().add(Ecdar.CANVAS_PADDING * 3);
     }
 
     @Override
     public ObservableValue<? extends Number> getEdgeY() {
-        return new SimpleDoubleProperty(Grid.TOOL_BAR_HEIGHT + 2 * Grid.GRID_SIZE - 4d);
+        return new SimpleDoubleProperty(Ecdar.CANVAS_PADDING * 4 - 4d);
```
Comment (seblund) [inline src/main/java/ecdar/abstractions/SystemRoot.java:41]: Are we sure we don't want to keep variables for the toolbar height (`TOOL_BAR_HEIGHT`) and corner size (`CORNER_SIZE`). It seems like they were used often before and now they are replaced by magical constants, are they no longer useful without the grid?

File: src/main/java/ecdar/model_canvas/arrow_heads/SimpleArrowHead.java [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
 import ecdar.utility.Highlightable;
 import ecdar.utility.helpers.SelectHelper;
 import javafx.scene.shape.Line;
 
 public class SimpleArrowHead extends ArrowHead implements SelectHelper.Selectable, Highlightable {
 
-    private static final double TRIANGLE_LENGTH = Grid.GRID_SIZE * 1.5;
-    private static final double TRIANGLE_WIDTH = Grid.GRID_SIZE;
+    private static final double TRIANGLE_LENGTH = 15;
+    private static final double TRIANGLE_WIDTH = 10;
```
Comment (seblund) [inline src/main/java/ecdar/model_canvas/arrow_heads/SimpleArrowHead.java:11]: Consider using `Ecdar.CANVAS_PADDING`

File: src/main/java/ecdar/presentations/ComponentInstancePresentation.java [hunk 4]
```diff
@@ -352,13 +351,13 @@ public Color.Intensity getColorIntensity() {
...
+        final ObservableDoubleValue minX = new SimpleDoubleProperty(Ecdar.CANVAS_PADDING);
         final ObservableDoubleValue maxX = controller.getSystem().getBox().getWidthProperty()
-                .subtract(GRID_SIZE)
+                .subtract(Ecdar.CANVAS_PADDING)
                 .subtract(controller.getInstance().getBox().getWidth());
-        final ObservableDoubleValue minY = new SimpleDoubleProperty(Grid.TOOL_BAR_HEIGHT + Grid.GRID_SIZE * 3);
+        final ObservableDoubleValue minY = new SimpleDoubleProperty(Ecdar.CANVAS_PADDING * 5);
         final ObservableDoubleValue maxY = controller.getSystem().getBox().getHeightProperty()
-                .subtract(GRID_SIZE)
+                .subtract(Ecdar.CANVAS_PADDING * 5)
```
Comment (seblund) [inline src/main/java/ecdar/presentations/ComponentInstancePresentation.java:360]: This is 5x the value it was previously, is this an error?

File: src/main/java/ecdar/presentations/ComponentOperatorPresentation.java [hunk 5]
```diff
@@ -146,13 +147,13 @@ public Color.Intensity getColorIntensity() {
...
+        final ObservableDoubleValue minX = new SimpleDoubleProperty(Ecdar.CANVAS_PADDING);
         final ObservableDoubleValue maxX = controller.getSystem().getBox().getWidthProperty()
-                .subtract(Grid.GRID_SIZE)
+                .subtract(Ecdar.CANVAS_PADDING)
                 .subtract(controller.getOperator().getBox().getWidth());
-        final ObservableDoubleValue minY = new SimpleDoubleProperty(Grid.TOOL_BAR_HEIGHT + Grid.GRID_SIZE * 3);
+        final ObservableDoubleValue minY = new SimpleDoubleProperty(Ecdar.CANVAS_PADDING * 5);
         final ObservableDoubleValue maxY = controller.getSystem().getBox().getHeightProperty()
-                .subtract(Grid.GRID_SIZE)
+                .subtract(Ecdar.CANVAS_PADDING * 5)
```
Comment (seblund) [inline src/main/java/ecdar/presentations/ComponentOperatorPresentation.java:156]: Same as before: This is 5x the value it was previously, is this an error?

File: src/main/java/ecdar/presentations/ComponentPresentation.java [hunk 6]
```diff
@@ -252,15 +251,15 @@ ModelController getModelController() {
     @Override
     double getDragAnchorMinHeight() {
         final Component component = controller.getComponent();
-        double minHeight = 10 * GRID_SIZE;
+        double minHeight = 100;
```
Comment (seblund) [inline src/main/java/ecdar/presentations/ComponentPresentation.java:254]: Consider making a function of `Ecdar.CANVAS_PADDING`

File: src/main/java/ecdar/presentations/ModelPresentation.java [hunk 7]
```diff
@@ -87,7 +85,7 @@ private void initializeName() {
         updateColor.run();
 
         // Center the text vertically and aff a left padding of CORNER_SIZE
-        controller.name.setPadding(new Insets(2, 0, 0, Grid.CORNER_SIZE));
+        controller.name.setPadding(new Insets(2, 0, 0, 40));
```
Comment (seblund) [inline src/main/java/ecdar/presentations/ModelPresentation.java:88]: Comment is outdated and consider a function of `Ecdar.CANVAS_PADDING`

File: src/main/java/ecdar/presentations/SystemPresentation.java [hunk 8]
```diff
@@ -79,8 +78,8 @@ private void initializeFrame() {
         // Generate top right corner (to subtract)
         final Polygon topRightCorner = new Polygon(
                 system.getBox().getWidth(), 0,
-                system.getBox().getWidth() - Grid.CORNER_SIZE - 2, 0,
-                system.getBox().getWidth(), Grid.CORNER_SIZE + 2
+                system.getBox().getWidth() - Ecdar.CANVAS_PADDING * 3 + 8, 0,
```
Comment (seblund) [inline src/main/java/ecdar/presentations/SystemPresentation.java:81]: I feel like this makes more sense if canvas padding changes
```suggestion
                system.getBox().getWidth() - Ecdar.CANVAS_PADDING * 4 - 2, 0,
```

File: src/main/java/ecdar/presentations/SystemRootPresentation.java [hunk 9]
```diff
@@ -40,12 +40,12 @@ private void initializeDimensions() {
         setLayoutX(root.getX());
         root.getXProperty().bind(layoutXProperty());
 
-        setLayoutY(Grid.TOOL_BAR_HEIGHT);
+        setLayoutY(40);
```
Comment (seblund) [inline src/main/java/ecdar/presentations/SystemRootPresentation.java:43]: Consider making the constants in this file a function of `Ecdar.CANVAS_PADDING`

File: src/main/java/ecdar/presentations/TagPresentation.java [hunk 10]
```diff
@@ -40,7 +37,7 @@ public class TagPresentation extends StackPane {
     LineTo l3;
     boolean hadInitialFocus = false;
 
-    static double TAG_HEIGHT = 1.6 * GRID_SIZE;
+    static double TAG_HEIGHT = 16;
```
Comment (seblund) [inline src/main/java/ecdar/presentations/TagPresentation.java:40]: Consider making this a function of `Ecdar.CANVAS_PADDING`

File: src/main/java/ecdar/utility/keyboard/NudgeDirection.java [hunk 11]
```diff
@@ -1,12 +1,10 @@
...
-
 public enum NudgeDirection {
-    UP(0, -GRID_SIZE),
-    RIGHT(GRID_SIZE, 0),
-    DOWN(0, GRID_SIZE),
-    LEFT(-GRID_SIZE, 0);
+    UP(0, -10),
+    RIGHT(10, 0),
+    DOWN(0, 10),
+    LEFT(-10, 0);
```
Comment (seblund) [inline src/main/java/ecdar/utility/keyboard/NudgeDirection.java:7]: Consider making these a function of `Ecdar.CANVAS_PADDING`

**Review by seblund — COMMENTED** [review]
I'm not super familiar with the GUI code base but I went through the changes and they look good to me. 

Generally I was a little uncertain as to when `Ecdar.CANVAS_PADDING` should be used instead of constants as I think the named constants should be preferred for readability. I've made comments where I would think that either the canvas padding or a new constant variable should be used. You can just let me know where it doesn't make sense :)

File: src/main/java/ecdar/abstractions/ComponentInstance.java (hunk 1 above)
Comment (Nielswps) [reply src/main/java/ecdar/abstractions/ComponentInstance.java:13]: This introduces a dependency between the padding and component size, which can result in unforeseen side-effects if the padding is changed. There is no real connection between the padding and the component sizes IMO

File: src/main/java/ecdar/model_canvas/arrow_heads/SimpleArrowHead.java (hunk 3 above)
Comment (Nielswps) [reply src/main/java/ecdar/model_canvas/arrow_heads/SimpleArrowHead.java:11]: Changing the padding would change the size of the arrow head, which IMO is unexpected

File: src/main/java/ecdar/presentations/ComponentOperatorPresentation.java (hunk 5 above)
Comment (Nielswps) [reply src/main/java/ecdar/presentations/ComponentOperatorPresentation.java:156]: Changed to `Ecdar.CANVAS_PADDING`, but has no effect

File: src/main/java/ecdar/model_canvas/arrow_heads/SimpleArrowHead.java (hunk 3 above)
Comment (Nielswps) [reply src/main/java/ecdar/model_canvas/arrow_heads/SimpleArrowHead.java:11]: Changing the padding would change the size of the arrow head, which IMO is unexpected

File: src/main/java/ecdar/presentations/ComponentPresentation.java (hunk 6 above)
Comment (Nielswps) [reply src/main/java/ecdar/presentations/ComponentPresentation.java:254]: Changing the padding would have, IMO, unexpected side effect here

File: src/main/java/ecdar/presentations/ModelPresentation.java (hunk 7 above)
Comment (Nielswps) [reply src/main/java/ecdar/presentations/ModelPresentation.java:88]: `PADDING` constant added

File: src/main/java/ecdar/presentations/SystemPresentation.java (hunk 8 above)
Comment (Nielswps) [reply src/main/java/ecdar/presentations/SystemPresentation.java:81]: `CORNER_SIZE` constant introduced for models

File: src/main/java/ecdar/presentations/SystemRootPresentation.java (hunk 9 above)
Comment (Nielswps) [reply src/main/java/ecdar/presentations/SystemRootPresentation.java:43]: `TOOLBAR_HEIGHT` constant introduced

File: src/main/java/ecdar/presentations/TagPresentation.java (hunk 10 above)
Comment (Nielswps) [reply src/main/java/ecdar/presentations/TagPresentation.java:40]: IMO unexpected dependency

File: src/main/java/ecdar/utility/keyboard/NudgeDirection.java (hunk 11 above)
Comment (Nielswps) [reply src/main/java/ecdar/utility/keyboard/NudgeDirection.java:7]: IMO unexpected dependecy

**Review by seblund — APPROVED** [review]
Looks good to me!

**Pull request merged**